### PR TITLE
Fix faulty unique symbol check

### LINKED ISSUE #17992: Incorrect redeclared symbol check logic for lambdas
The below check is incorrect. The arguments should be swapped and the flags of the owner of the symbol should be checked.
https://github.com/ballerina-platform/ballerina-lang/blob/df71d42b7ccf294e8804f053ba8f8d024ab3f8ca/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java#L262

### PR DESCRIPTION
**Pull request by pubudu91**
Fixes #17992

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 1]
```diff
@@ -168,15 +168,15 @@ public boolean checkForUniqueSymbol(DiagnosticPos pos, SymbolEnv env, BSymbol sy
         }
 
         // if a symbol is found, then check whether it is unique
-        return isSameSymbol(pos, symbol, foundSym);
+        return !isDuplicateSymbol(pos, symbol, foundSym);
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java]: All the usage of `isDuplicateSymbol` starts with a `!` hence, Consider renaming this.

**Comment by hasithaa** [comment]
Shall we add test cases as well?

**Comment by pubudu91** [comment]
The test cases are already there in https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/resources/test-src/variable/shadowing/variable-shadowing-negative.bal

It's just that even with the previous incorrect check, one couldn't write code in a way which went in the wrong branch of the check. It would have had to be something like below:
```ballerina
public function main(string... args) {
    string '\$lambda\$0 = "foo";

    var fn = function () returns string {
        return "bar";
    };
}
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java (hunk 1 above)
Comment (pubudu91) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java]: renamed and inverted to isDistinctSymbol()

**Pull request merged**